### PR TITLE
Use async logging in real-time critical code

### PR DESCRIPTION
**Pull request by ashleyz**

### CODE REVIEW

**Comment by padenot** [comment]
@ashleyz This looks like it could be merged, right?

You can pull https://github.com/mozilla/cubeb-pulse-rs/commit/ab3c0239ad490ce9d12cd7a68d714fb19c9b1004 here to fix Clippy on Nightly.

File: src/backend/context.rs [hunk 1]
```diff
@@ -305,7 +305,7 @@ impl ContextOps for PulseContext {
             user_data: *mut c_void,
         ) {
             let list_data = unsafe { &mut *(user_data as *mut PulseDevListData) };
-            let ctx = &(*list_data.context);
+            let ctx = list_data.context;
```
Comment (ashleyz) [inline src/backend/context.rs:308]: This change is to fix:
```
$  rustup run nightly cargo clippy --all -- -D warnings
error: deref on an immutable reference
   --> src/backend/context.rs:309:11
    |
309 | let ctx = &(*list_data.context);
    |           ^^^^^^^^^^^^^^^^^^^^^ help: if you would like to reborrow, try removing `&*`: `list_data.context`
```

File: src/backend/context.rs [hunk 2]
```diff
@@ -366,7 +366,7 @@ impl ContextOps for PulseContext {
...
```
Comment (ashleyz) [inline src/backend/context.rs:369]: Fix clippy error

File: src/backend/context.rs [hunk 3]
```diff
@@ -524,7 +524,7 @@ impl ContextOps for PulseContext {
         user_ptr: *mut c_void,
     ) -> Result<Stream> {
         if self.error {
-            let _ = self.context_init()?;
+            self.context_init()?;
```
Comment (ashleyz) [inline src/backend/context.rs:527]: Fix clippy error

File: src/backend/context.rs [hunk 4]
```diff
@@ -437,7 +437,7 @@ impl ContextOps for PulseContext {
                     .unwrap_or_default();
             }
 
-            (*list_data.context).mainloop.signal();
+            list_data.context.mainloop.signal();
```
Comment (ashleyz) [inline src/backend/context.rs:440]: Fix clippy error

File: src/backend/context.rs (hunk 1 above)
Comment (ChunMinChang) [reply src/backend/context.rs:308]: it would be great if these clippy fixes are separated into different commits.

**Review by ChunMinChang — APPROVED** [review]
LGTM, thanks!

**Pull request merged**